src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -387,7 +387,7 @@ public sealed override bool Equals(object symbol)
                 && lambda._syntax == _syntax
                 && lambda._refKind == _refKind
                 && lambda.ReturnType.TypeSymbol == this.ReturnType.TypeSymbol
-                && System.Linq.ImmutableArrayExtensions.SequenceEqual(lambda.ParameterTypes, this.ParameterTypes)
+                && System.Linq.ImmutableArrayExtensions.SequenceEqual(lambda.ParameterTypes, this.ParameterTypes, (p1, p2) => p1.TypeSymbol.Equals(p2.TypeSymbol))
```

Comment:
Not for the PR: consider implementing `IEquatable` on `TypeSymbolWithAnnotations`